### PR TITLE
2834967 Support all purchasable entities

### PR DESCRIPTION
**Pull request by olafkarsten**
[issue 2834967](https://www.drupal.org/node/2834967) Support all purchasable entities

Most of this was renaming of variables, removing type checks and so on.

Requiered some schema changes, so you have to reinstall commerce_stock_local.
`drush devel-reinstall -y commerce_stock_local`

It's not done yet. We have to refactor the cron stuff and the transactions code.  But we need to agree on the "how" first. There sould be some issues open. So we can first go with this PR and do the remaining parts later.

### CODE REVIEW

**Comment by olafkarsten** [comment]
@BBGuy You can probably [squash](https://help.github.com/articles/about-pull-request-merges/) all the commits into a single one. So we keep a cleaner history in the main repo.

File: modules/local_storage/src/Plugin/QueueWorker/StockLevelUpdater.php [hunk 1]
```diff
@@ -12,18 +12,20 @@
  *   title = @Translation("Commerce Stock Local stock level updater"),
  *   cron = {"time" = 10}
  * )
+ *
```
Comment (steveoliver) [inline modules/local_storage/src/Plugin/QueueWorker/StockLevelUpdater.php:15]: @todo Inject the service from the container through __construct()/::create()

**Review by steveoliver — COMMENTED** [review]
Looks great - I would say lets add tests, i.e. for getIsStockManaged() but since that currently is not implemented (always returns TRUE), I don't see the point in this issue.

**Comment by BBGuy** [comment]
Great work. pushed to drupal.org and github: https://github.com/BBGuy/commerce_stock/commit/76237c0e81f3370c24a77142d23cb875049f9600

**Pull request closed without merging**